### PR TITLE
Display port visits as one event (entry and exit) in Vessel Viewer for insurers. VV-426

### PR DESCRIPTION
**Pull request by rdgfuentes**
https://globalfishingwatch.atlassian.net/browse/VV-426
image

### CODE REVIEW

File: apps/vessel-history/features/activity-by-type/activity-by-type.selectors.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import { createSelector } from '@reduxjs/toolkit'
+import { DateTime } from 'luxon'
+import { t } from 'features/i18n/i18n'
+import { selectFilteredEvents } from 'features/vessels/activity/vessels-activity.selectors'
+import { PortVisitSubEvent } from 'types/activity'
+import { EventTypes } from '../../../../libs/api-types/src/events'
```
Comment (fertandil87) [inline apps/vessel-history/features/activity-by-type/activity-by-type.selectors.ts:6]: the path..

File: apps/vessel-history/features/activity-by-type/activity-by-type.selectors.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+      const isPortExit = event.portVisitSubEvent === PortVisitSubEvent.Exit
+      const isPortEntry = event.portVisitSubEvent === PortVisitSubEvent.Entry
+      const id = event.id.split('-').shift()
+      const portEntry =
+        (isPortEntry && event) || events.find((ev) => ev.id === `${id}-${PortVisitSubEvent.Entry}`)
+      const portExit =
+        (isPortExit && event) || events.find((ev) => ev.id === `${id}-${PortVisitSubEvent.Exit}`)
+
+      const durationDiff = DateTime.fromMillis(portExit.end as number).diff(
```
Comment (fertandil87) [inline apps/vessel-history/features/activity-by-type/activity-by-type.selectors.ts:22]: do we need this if we have the property durationHrs?

Comment (fertandil87) [reply apps/vessel-history/features/activity-by-type/activity-by-type.selectors.ts:22]: also you already have a property durationDescription

File: apps/vessel-history/features/activity-by-type/activity-by-type.selectors.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+export const selectFilteredEventsWithMainPortVisit = createSelector(
+  [selectFilteredEvents],
+  (events) =>
+    events.reduce((previous, event) => {
+      if (event.type !== EventTypes.Port) return [...previous, event]
+
+      const isPortExit = event.portVisitSubEvent === PortVisitSubEvent.Exit
+      const isPortEntry = event.portVisitSubEvent === PortVisitSubEvent.Entry
+      const id = event.id.split('-').shift()
+      const portEntry =
```
Comment (fertandil87) [inline apps/vessel-history/features/activity-by-type/activity-by-type.selectors.ts:17]: in the event object you already have the start and end, you dont need the finds

**Review by fertandil87 — CHANGES_REQUESTED** [review]

File: apps/vessel-history/features/activity-by-type/activity-by-type.selectors.ts (hunk 3 above)
Comment (rdgfuentes) [reply apps/vessel-history/features/activity-by-type/activity-by-type.selectors.ts:17]: In this case the start property of the exit port visit was already overwritten (https://github.com/GlobalFishingWatch/frontend/blob/feature/VV-426/apps/vessel-history/features/resources/resources.slice.ts#L54) so at least the port entry will have to stay.

File: apps/vessel-history/features/activity-by-type/activity-by-type.selectors.ts (hunk 2 above)
Comment (rdgfuentes) [reply apps/vessel-history/features/activity-by-type/activity-by-type.selectors.ts:22]: Same as above, given that the start property of the exit port visit was already overwritten the duration in this case is always 0, thus we have to calculate it.

**Pull request merged**